webrender/src/resource_cache.rs

```diff
@@ -621,18 +601,40 @@ impl Resource for CachedImageInfo {
     }
 }
 
-fn spawn_glyph_cache_thread() -> (Sender<GlyphCacheMsg>, Receiver<GlyphCacheResultMsg>) {
-    // Used for messages from resource cache -> glyph cache thread.
+
+/// Message sent from the resource cache to the vector cache thread.
+pub enum VectorCacheMsg {
+    /// Begin the frame - pass ownership of the vector cache to the thread.
+    BeginFrame(FrameId, GlyphCache),
+    /// Add a new font.
+    AddFont(FontKey, FontTemplate),
+    /// Request glyphs for a text run.
+    RequestGlyphs(FontKey, Au, ColorF, Vec<GlyphInstance>, FontRenderMode, Option<GlyphOptions>),
+    /// Finished requesting glyphs. Reply with new glyphs.
+    EndFrame(Sender<VectorCacheResultMsg>),
+}
+
+/// Results send from glyph cache thread back to main resource cache.
+pub enum VectorCacheResultMsg {
+    /// Return the glyph cache, and a list of newly rasterized glyphs.
+    EndFrame(GlyphCache, Vec<GlyphRasterJob>),
+}
+
+pub struct GlyphRasterJob {
+    pub key: RenderedGlyphKey,
+    pub result: Option<RasterizedGlyph>,
+}
+
+pub fn spawn_vector_cache_thread(worker_count: usize) -> Sender<VectorCacheMsg> {
+    let worker_count = if worker_count > 0 { worker_count } else { 1 };
```

Comment:
I assume we'll have something like `const DEFAULT_WORKER_COUNT: usize = 1` eventually